script.js

```diff
@@ -1,12 +1,261 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+// Cache for fetched data
+const cache = {
+  shows: null,
+  episodes: {},
+  firstFetch: false,
+};
```

Comment:
Just amazing 🤩 This approach shows your understanding is way above your level. Well done!